src/Dialect/ONNX/Transforms/ConstProp.cpp

```diff
@@ -203,7 +254,41 @@ struct ElementWiseBinaryOpImpl<ONNXMulOp, T, EnableNotBool<T>> {
 
 template <typename T>
 struct ElementWiseBinaryOpImpl<ONNXDivOp, T, EnableNotBool<T>> {
-  static T eval(T lhs, T rhs) { return lhs / rhs; }
+  static T eval(T lhs, T rhs) {
+    if constexpr (std::is_integral_v<T>) {
+      if (rhs == 0) {
+        // Undefined behavior. We can return any value.
+        // Performing the divison would crash.
+        return lhs;
```

Comment:
I prefer to stop compilation here since this is actually an error. The input model is not good or the compiler produces this zero. 

@AlexandreEichenberger any thought?

Reply:
I guess this also relates to my previous comment when `Min > Max`.

Reply:
ONNX unfortunately doesn't specify what the behavior should be; whether the model is illegal and should be rejected or whether just the output of that operator is unspecified, but the rest of the model continues (and potentially conditionally doesn't use the result).

Reply:
Does anyone know how onnxruntime handle this situation?

I just checked how the current generated code for non-constant inputs deals with div-by-zero on a Z machine. The program does not crash and these are the results:
- f32 data type: the output is `negative infinity` if `rhs` is negative, `positive infinity` if `rhs` is positive. This result may be different on other architectures.
- integer type: the output is kind of random. [According to MLIR documentation](https://mlir.llvm.org/docs/Dialects/ArithOps/#arithdivsi-arithdivsiop), the div-by-zero for integer is undefined behaviour.


Anyway, it's constant propagation and it's useful if we would print out a warning message when we see div-by-zero.

Reply:
I responded to the (https://github.com/onnx/onnx/issues/6165) PR. I believe this situation should be declared 'undefined' and we should specify in the specs that tools (such as ORT, onnx-mlir) should individually define what they do in such situations.

In our case (onnx-mlir), I think it would be reasonable to generate a user error and abort the compilation when the error can be detected at compile time, and silently ignore it if the values are not known at compile time. If desired, we could implement a "fussy" compilation mode where such assumptions are tested at runtime by adding some dynamic asserts. This would require some new infra as I don't think we have at this time a runtime method that provide a "assert if this condition is false". Obviously, it would not necessarily be much work to add such a runtime operation.